internal/resolution/variable_sources/global_constraints/global_constraints_test.go

```diff
@@ -0,0 +1,313 @@
+			}),
+			required_package.NewRequiredPackageVariable("test-package-2", []*olmentity.BundleEntity{
+				olmentity.NewBundleEntity(bundleSet["bundle-15"]),
+				olmentity.NewBundleEntity(bundleSet["bundle-16"]),
+				olmentity.NewBundleEntity(bundleSet["bundle-17"]),
+			}),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-2"]),
+				[]*olmentity.BundleEntity{
+					olmentity.NewBundleEntity(bundleSet["bundle-4"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-5"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-7"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-8"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-9"]),
+				},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-1"]),
+				[]*olmentity.BundleEntity{
+					olmentity.NewBundleEntity(bundleSet["bundle-7"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-8"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-9"]),
+				},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-4"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-5"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-6"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-7"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-8"]),
+				[]*olmentity.BundleEntity{
+					olmentity.NewBundleEntity(bundleSet["bundle-9"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-10"]),
+					olmentity.NewBundleEntity(bundleSet["bundle-11"]),
+				},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-9"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-10"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-11"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-15"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-16"]),
+				[]*olmentity.BundleEntity{},
+			),
+			bundles_and_dependencies.NewBundleVariable(
+				olmentity.NewBundleEntity(bundleSet["bundle-17"]),
+				[]*olmentity.BundleEntity{},
+			),
+		}
+		variables, err := globalConstraintVariableSource.GetVariables(ctx, entitySource)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(variables)).To(Equal(26))
```

Comment:
How do we land on 26 variables? I counted 16 bundles defined in the `bundleSet` and I see that we have 11 globalConstraintVariables below, which adds up to 27.

Reply:
And I keep getting a different count‽‽

Reply:
oh oh

Reply:
ah ok - so, it receives 15 variables upstream from it and produces another 11. It seems to be correct.
Basically, you have the required package variables, the bundle variables, and it spits out the uniqueness variables.